airflow/providers/sftp/operators/sftp.py

```diff
@@ -88,6 +124,9 @@ def __init__(
         remote_host=None,
         local_filepath=None,
         remote_filepath=None,
+        local_folder=None,
+        remote_folder=None,
+        regexp_mask=None,
```

Comment:
I am really not fond of this argument list since it’s too easy to pass in nonsensical combinations e.g.

```python
SFTPOperator(
    local_filepath="/my/file",
    local_folder="/my/dir",
)
```

But I understand this is not easy to change the arguments due to compatibility considerations.

Perhapes what we should do is to introduce a separate operator (say `SFTPBatchOperator`) to handle the new feature, instead of bolting it onto the existing one.

Reply:
for this case will print AirflowException with examples
https://github.com/apache/airflow/pull/16792/files#diff-911f39c07d851bc2f0a2958674ed940ddc68aea6ecc7d71c29567816c182eaf6R216

I checked the example of dag again, I am comfortable working with one operetor, even if he has combined arguments.

Your idea i understand too, may be we will wait @potiuk and hear his opinion